packages/core-data/src/test/reducer.js

```diff
@@ -99,6 +107,206 @@ describe( 'entities', () => {
 	} );
 } );
 
+describe( 'undo', () => {
+	let lastEdits;
+	let undoState;
+	let expectedUndoState;
+	const createEditActionPart = ( edits ) => ( {
+		kind: 'someKind',
+		name: 'someName',
+		recordId: 'someRecordId',
+		edits,
+	} );
+	const createNextEditAction = ( edits, transientEdits = {} ) => {
+		let action = {
+			...createEditActionPart( edits ),
+			transientEdits,
+		};
+		action = {
+			type: 'EDIT_ENTITY_RECORD',
```

Comment:
I really liked how the previous history reduces could work with any state and actions. This one seems to handle `EDIT_ENTITY_RECORD`. Is there any way it could be more abstracted? Cc @aduth.

Reply:
What other actions does it need to handle for entities?

I don't think this is how this reducer will look like in the long term, specially after adding support for collaborative editing.

But, for now it gives us a lot of control over details that allow us to meet requirements for things like explicit levels and other nuances that the editor and block editor need. A more generalized abstraction could make that harder.